subsys/net/ip/connection.c

```diff
@@ -515,173 +523,228 @@ static enum net_verdict conn_raw_socket(struct net_pkt *pkt,
 
 		if ((type == SOCK_DGRAM && !net_pkt_is_l2_processed(pkt)) ||
 		    (type == SOCK_RAW && net_pkt_is_l2_processed(pkt))) {
-			goto out;
+			return NET_CONTINUE;
 		}
 	}
 
-	if (conn->flags & NET_CONN_LOCAL_ADDR_SET) {
-		struct net_if *pkt_iface = net_pkt_iface(pkt);
-		struct sockaddr_ll *local;
-		struct net_pkt *raw_pkt;
+	if (!(conn->flags & NET_CONN_LOCAL_ADDR_SET)) {
+		return NET_CONTINUE;
+	}
 
-		local = (struct sockaddr_ll *)&conn->local_addr;
+	struct net_if *pkt_iface = net_pkt_iface(pkt);
+	struct sockaddr_ll *local;
+	struct net_pkt *raw_pkt;
 
-		if (local->sll_ifindex !=
-		    net_if_get_by_iface(pkt_iface)) {
-			return NET_CONTINUE;
-		}
+	local = (struct sockaddr_ll *)&conn->local_addr;
 
-		NET_DBG("[%p] raw match found cb %p ud %p", conn,
-			conn->cb, conn->user_data);
+	if (local->sll_ifindex != net_if_get_by_iface(pkt_iface)) {
+		return NET_CONTINUE;
+	}
 
-		raw_pkt = net_pkt_clone(pkt, CLONE_TIMEOUT);
-		if (!raw_pkt) {
-			net_stats_update_per_proto_drop(pkt_iface, proto);
-			NET_WARN("pkt cloning failed, pkt %p dropped", pkt);
-			return NET_DROP;
-		}
+	NET_DBG("[%p] raw match found cb %p ud %p", conn, conn->cb,
+		conn->user_data);
 
-		if (conn->cb(conn, raw_pkt, NULL, NULL, conn->user_data)
-		    == NET_DROP) {
-			net_stats_update_per_proto_drop(pkt_iface, proto);
-			net_pkt_unref(raw_pkt);
-		} else {
-			net_stats_update_per_proto_recv(pkt_iface, proto);
-		}
+	raw_pkt = net_pkt_clone(pkt, CLONE_TIMEOUT);
+	if (!raw_pkt) {
+		net_stats_update_per_proto_drop(pkt_iface, proto);
+		NET_WARN("pkt cloning failed, pkt %p dropped", pkt);
+		return NET_DROP;
+	}
 
-		return NET_OK;
+	if (conn->cb(conn, raw_pkt, NULL, NULL, conn->user_data) == NET_DROP) {
+		net_stats_update_per_proto_drop(pkt_iface, proto);
+		net_pkt_unref(raw_pkt);
+	} else {
+		net_stats_update_per_proto_recv(pkt_iface, proto);
 	}
 
-out:
-	return NET_CONTINUE;
+	return NET_OK;
 }
 
 enum net_verdict net_conn_input(struct net_pkt *pkt,
 				union net_ip_header *ip_hdr,
 				uint8_t proto,
 				union net_proto_header *proto_hdr)
 {
-	struct net_if *pkt_iface = net_pkt_iface(pkt);
```

Comment:
While not a strict requirement, we tend to declare local variables at the beginning of the block, I wouldn't change that, unless necessary.

Reply:
Yes you're right - reverted. This was just a leftover of me trying to identify opportunities to improve code encapsulation (e.g. introducing smaller sub-functions with less local variables).